eo-maven-plugin/src/main/java/org/eolang/maven/name/OnCached.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2016-2023 Objectionary.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.eolang.maven.name;
+
+import org.cactoos.scalar.Sticky;
+import org.cactoos.scalar.Unchecked;
+import org.eolang.maven.hash.CommitHash;
+
+/**
+ * Cached object name.
+ *
+ * @since 0.29.6
+ */
+public final class OnCached implements ObjectName {
+
+    /**
+     * Value.
+     */
+    private final Unchecked<String> vle;
+
+    /**
+     * Hash.
+     */
+    private final Unchecked<CommitHash> hsh;
+
+    /**
+     * Self.
+     */
+    private final Unchecked<String> self;
+
+    /**
+     * Ctor.
+     * @param src Origin.
+     */
+    public OnCached(final ObjectName src) {
+        this.vle = new Unchecked<>(new Sticky<>(src::value));
+        this.hsh = new Unchecked<>(new Sticky<>(src::hash));
+        this.self = new Unchecked<>(new Sticky<>(src::asString));
+    }
+
+    @Override
+    public String value() {
+        return this.vle.value();
+    }
+
+    @Override
+    public CommitHash hash() {
+        return this.hsh.value();
+    }
+
+    @Override
+    public String asString() {
+        return this.self.value();
+    }
+
+    @Override
+    public String toString() {
```

Comment:
@maxonfjvipon Do we use that method somewhere?

Reply:
@volodya-lombrozo see [this comment](https://github.com/objectionary/eo/pull/2334#discussion_r1280267116)